sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/CustomClaimRequest.java

```diff
@@ -0,0 +1,80 @@
+package com.azure.identity.implementation;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.microsoft.aad.msal4j.AuthenticationErrorCode;
+import com.microsoft.aad.msal4j.ClaimsRequest;
+import com.microsoft.aad.msal4j.MsalClientException;
+import com.microsoft.aad.msal4j.RequestedClaimAdditionalInfo;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+
+public class CustomClaimRequest extends ClaimsRequest {
```

Comment:
Temporary impl class, until MSAL releases support to convert jsonString claims to ClaimsRequest class.